ssl/quic/quic_impl.c

```diff
@@ -3571,6 +3571,27 @@ const SSL_CIPHER *ossl_quic_get_cipher(unsigned int u)
     return NULL;
 }
 
+/*
+ * SSL_get_shutdown()
+ * ------------------
+ */
+int ossl_quic_get_shutdown(const SSL *s)
+{
+    QCTX ctx;
+    int shut = 0;
+
+    if (!expect_quic_conn_only(s, &ctx))
+        return 0;
+
+    if (ossl_quic_channel_is_term_any(ctx.qc->ch)) {
+        shut |= SSL_SENT_SHUTDOWN;
```

Comment:
Is this really correct? It seems like we should vary it based on whether the TCAUSE was local or remote. Depending on that these flags should appear in the opposite chronological order based on these two predicates.

Reply:
IMO it is correct. When we receive the CONNECTION_CLOSE frame we immediately reply with connection close too and enter the DRAINING state. I do not see how to implement it otherwise.